pkg/objcache/objcache.go

```diff
@@ -77,14 +77,16 @@ var ErrCacheFull = errors.New("Not enough space in cache")
 // Used for adding entry to the object cache. Implements io.WriteCloser
 type cacheBuffer struct {
 	*bytes.Buffer // Implements io.Writer
-	onClose       func()
+	onClose       func() error
 }
 
 // On close, onClose() is called which checks if all object contents
 // have been written so that it can save the buffer to the cache.
-func (c cacheBuffer) Close() error {
-	c.onClose()
-	return nil
+func (c cacheBuffer) Close() (err error) {
+	err = c.onClose()
+	// Reset the buffer since we are not going to use it anymore.
```

Comment:
better not do c.Reset() because it indicates that we are going to re-use the buffer.